common/src/main/java/org/opensearch/sdk/client/LocalClusterIndicesClient.java

```diff
@@ -214,6 +220,99 @@ public CompletionStage<DeleteDataObjectResponse> deleteDataObjectAsync(
         }), executor);
     }
 
+    @Override
+    public CompletionStage<BulkDataObjectResponse> bulkDataObjectAsync(
+        BulkDataObjectRequest request,
+        Executor executor,
+        Boolean isMultiTenancyEnabled
+    ) {
+        return CompletableFuture.supplyAsync(() -> AccessController.doPrivileged((PrivilegedAction<BulkDataObjectResponse>) () -> {
+            try {
+                log.info("Performing {} bulk actions on indices {}", request.requests().size(), request.getIndices());
+                BulkRequest bulkRequest = new BulkRequest();
+
+                for (DataObjectRequest dataObjectRequest : request.requests()) {
+                    if (dataObjectRequest instanceof PutDataObjectRequest) {
+                        try (XContentBuilder sourceBuilder = XContentFactory.jsonBuilder()) {
```

Comment:
> Should we add any retry mechanism for bulk requests?

In the case of the Local and Remote clients, we are just delegating exactly to the existing NodeClient version of this.  So we would only add retries on the individual requests if they already existed in the code.

In the case of the two places where we are implementing this for undeploying a model, we are only using the update request and we are not doing retries there.  If there's a retry on the update request it's handled here, but it doesn't look like that was actually done.  (When I implement this I'll make sure every option is handled.)

We can always add the capability in any of the requests later if it is needed; I have intentionally not tried to replicate every possible combination of parameters for the request objects until they are needed.

> during that time may be for a network glitches it failed to ingest one document. What happens that time?

1. That individual request would fail (the `failed` boolean would be set true in the `DataObjectResponse` subclass) and the JSON response could be decompiled into the full `BulkItemResponse` object with error information.
2. The caller would be able to test for any failures with a single check, and if there were, iterate the `BulkResponse` array to find the failed items.